public/app/plugins/panel/barchart/config.ts

```diff
@@ -0,0 +1,100 @@
+import { Dispatch, MutableRefObject, SetStateAction } from 'react';
+import { UPlotConfigBuilder } from '@grafana/ui';
+import { positionTooltip } from '@grafana/ui/src/components/uPlot/plugins/TooltipPlugin';
+import { CartesianCoords2D } from '@grafana/data';
+
+export type HoverEvent = {
+  xIndex: number;
+  yIndex: number;
+  pageX: number;
+  pageY: number;
+};
+
+type SetupConfigParams = {
+  config: UPlotConfigBuilder;
+  onUPlotClick: () => void;
+  setFocusedSeriesIdx: Dispatch<SetStateAction<number | null>>;
+  setFocusedPointIdx: Dispatch<SetStateAction<number | null>>;
+  setCoords: Dispatch<SetStateAction<CartesianCoords2D | null>>;
+  setHover: Dispatch<SetStateAction<HoverEvent | undefined>>;
+  isToolTipOpen: MutableRefObject<boolean>;
+};
+
+export const setupConfig = ({
```

Comment:
Maybe something like:
```suggestion
// This applies config hooks to setup tooltip listener.  Ideally this could happen in the same `prepConfig` function
// however the GraphNG structures do not allow access to the  `setHover` callback
export const setupTooltipConfig = ({
```